Services/RefundService.php

```diff
@@ -0,0 +1,281 @@
+                $this->partialRefundPayment($order, $molliePayment, $customAmount);
+            }
+        }
+
+        if ($transaction->getMollieOrderId() !== null) {
+            /** @var MollieOrder $mollieOrder */
+            $mollieOrder = $mollieClient->orders->get($transaction->getMollieOrderId());
+
+            if ($customAmount === null || $mollieOrder->amountCaptured->value <= $customAmount) {
+                $this->refundOrder($order, $mollieOrder);
+            }
+
+            if ($customAmount !== null && $mollieOrder->amountCaptured->value > $customAmount) {
+                throw new Exception('partial refunds of orders is currently not implemented.');
+            }
+        }
+    }
+
+    /**
+     * Refund a Mollie order
+     *
+     * @param Order       $order
+     * @param MollieOrder $mollieOrder
+     *
+     * @return bool|Refund
+     * @throws \Exception
+     *
+     */
+    public function refundOrder(Order $order, MollieOrder $mollieOrder)
+    {
+        $apiClient = $this->getMollieApi($order->getShop()->getId());
+
+        if (empty($this->modelManager) || $apiClient === null) {
+            return false;
+        }
+
+        /** @var OrderLinesRepository $mollieOrderLinesRepo */
+        $mollieOrderLinesRepo = $this->modelManager->getRepository(OrderLines::class);
+
+        $mollieShipmentLines = $mollieOrderLinesRepo->getShipmentLines($order);
+
+        /** @var Refund $refund */
+        $refund = $apiClient->orderRefunds->createFor(
+            $mollieOrder,
+            [
+                'lines' => $mollieShipmentLines
+            ]
+        );
+
+        if ($refund !== null) {
+            $this->processRefund($order);
+
+            if ($order->getDetails()->isEmpty() === false) {
+                foreach ($order->getDetails() as $detail) {
+                    $this->updateRefundedItemsOnOrderDetail($detail, $detail->getQuantity());
+                }
+            }
+        }
+
+        return $refund;
+    }
+
+    /**
+     * Refund a Mollie order
+     *
+     * @param Order       $order
+     * @param MollieOrder $mollieOrder
+     *
+     * @return bool|Refund
+     * @throws \Exception
+     *
+     */
+    public function partialRefundOrder(
+        Order $order,
+        Detail $detail,
+        MollieOrder $mollieOrder,
+        OrderLine $orderLine,
+        $quantity = 1
+    ) {
+        $apiClient = $this->getMollieApi($order->getShop()->getId());
+
+        if (empty($this->modelManager) || $apiClient === null) {
+            return false;
+        }
+
+        $data = [
+            'lines' => [
+                [
+                    'id' => $orderLine->id,
+                    'quantity' => $quantity,
+                ],
+            ]
+        ];
+
+        /** @var Refund $refund */
+        $refund = $apiClient->orderRefunds->createFor($mollieOrder, $data);
+
+        if ($refund !== null) {
```

Comment:
shouldnt there be an exception in here?
or is it totally fine that null will be returned? :)